src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Extract.cs

```diff
@@ -70,9 +70,12 @@ public void ExtractOutOfRoot(string entryName)
         [InlineData("NullCharFileName_FromWindows")]
         [InlineData("NullCharFileName_FromUnix")]
         [PlatformSpecific(TestPlatforms.AnyUnix)]  // Checks Unix-specific invalid file path
-        public void Unix_ZipWithInvalidFileNames_ThrowsArgumentException(string zipName)
+        public void Unix_ZipWithInvalidFileNames(string zipName)
         {
-            AssertExtensions.Throws<ArgumentException>("path", () => ZipFile.ExtractToDirectory(compat(zipName) + ".zip", GetTestFilePath()));
+            var testDirectory = GetTestFilePath();
```

Comment:
I still don't see what this test, and all the tests you modified in Zipfile.Create.cs, are doing that your new test isn't now doing. Why can't we delete all of those existing tests?

Reply:
I do not know _what_ they are doing per see and why they were added in the first place but I didn't want to remove them. I just fixed them and left them as is

Reply:
They were testing that extracting a zip file containing a file with a path with a null character, threw ArgumentException. It no longer does this, and your new test verifies that these cases are sanitized, so the tests have no value. Please remove these 3 tests

Reply:
Yep, I have the exact same code in the original test anyway.

Reply:
Are there 3 tests? I could only find two. One for windows and one for Unix

Reply:
Doh, you're right. 

Can this Unix test verify that the sanitization _didn't_ happen (except for the nulls)? Basically copy/paste exactly Windows_ZipWithInvalidFileNames, but change the expected strings.

Reply:
We have a test like that I think. `Unix_ZipWithOSSpecificFileNames` in Zipfile.Extract

Reply:
That only checks `<` and `>`. I don't have a strong opinion. Your call

Reply:
I'll sign off as-is 😄

Reply:
Well, we only sanitize nulls anyway, if the code worked as expected before, meaning we could extract zip files in UNIX with the correct result, the code we changed shouldn't affect that. We only added sanitization for null char in Unix and we test that already so I will say it is fine? I guess?